src/jemalloc.c

```
@@ -2183,6 +2183,9 @@ static bool
 malloc_init_hard(void) {
 	tsd_t *tsd;
 
+#ifdef LIMIT_USIZE_GAP
+	assert(TCACHE_MAXCLASS_LIMIT <= USIZE_GROW_SLOW_THRESHOLD);
```

It is also possible to assert in somewhere else like `tcache_init`, but putting it here could make it fail early. In this case, even when tcache is not enabled, if `TCACHE_MAXCLASS_LIMIT` is somehow set to a value too high, the assertion would still alarm.